omx/gstomxvideoenc.c

```diff
@@ -1464,15 +1529,147 @@ gst_omx_video_enc_fill_buffer (GstOMXVideoEnc * self, GstBuffer * inbuf,
+  if (!buffers) {
+    if (gst_omx_port_allocate_buffers (self->enc_in_port) != OMX_ErrorNone)
+      return FALSE;
+  } else {
+    if (gst_omx_port_use_buffers (self->enc_in_port, buffers) != OMX_ErrorNone) {
+      return FALSE;
+    }
+  }
+  /* If the output port is already configured with the new format we can
+   * allocate buffers here already */
+  if (self->enc_out_port->port_def.format.video.nFrameHeight == info->height
+      && self->enc_out_port->port_def.format.video.nFrameWidth == info->width) {
+    if (gst_omx_port_allocate_buffers (self->enc_out_port) != OMX_ErrorNone)
+      return FALSE;
+    have_output_buffers = TRUE;
+  } else {
+    /* Disable output port */
+    if (gst_omx_port_set_enabled (self->enc_out_port, FALSE) != OMX_ErrorNone)
+      return FALSE;
+
+    if (gst_omx_port_wait_enabled (self->enc_out_port,
+            1 * GST_SECOND) != OMX_ErrorNone)
+      return FALSE;
+  }
+
+  if (gst_omx_component_get_state (self->enc,
+          GST_CLOCK_TIME_NONE) != OMX_StateIdle)
+    return FALSE;
+
+  GST_DEBUG_OBJECT (self, "Changing state to Executing");
+  if (gst_omx_component_set_state (self->enc,
+          OMX_StateExecuting) != OMX_ErrorNone)
+    return FALSE;
+
+  if (gst_omx_component_get_state (self->enc,
+          GST_CLOCK_TIME_NONE) != OMX_StateExecuting)
+    return FALSE;
+
+  if (have_output_buffers) {
+    if (gst_omx_port_populate (self->enc_out_port) != OMX_ErrorNone)
+      return FALSE;
+    if (gst_omx_port_mark_reconfigured (self->enc_out_port) != OMX_ErrorNone)
+      return FALSE;
+  }
+
+  /* Start the srcpad loop again */
+  GST_DEBUG_OBJECT (self, "Starting out pad task");
+  self->downstream_flow_ret = GST_FLOW_OK;
+  gst_pad_start_task (GST_VIDEO_ENCODER_SRC_PAD (self),
+      (GstTaskFunction) gst_omx_video_enc_loop, self, NULL);
+
+  return TRUE;
+}
+
+static gboolean
+gst_omx_video_enc_use_buffers (GstOMXVideoEnc * self, GstOMXMemory * omxmem)
+{
+  GstOMXPort *mem_port;
+  OMX_PARAM_PORTDEFINITIONTYPE port_def;
+  GList *buffers = NULL;
+  gint i, n;
+  gboolean ret;
+
+  mem_port = omxmem->buf->port;
+  n = mem_port->buffers->len;
+
+  /* Set actual buffer count to the port */
+  gst_omx_port_get_port_definition (self->enc_in_port, &port_def);
+  port_def.nBufferCountActual = n;
+
+  GST_DEBUG_OBJECT (self, "Updating input port buffer count to %lu",
+      port_def.nBufferCountActual);
+  if (gst_omx_port_update_port_definition (self->enc_in_port,
+          &port_def) != OMX_ErrorNone)
+    goto reconfigure_error;
+
+  GST_DEBUG_OBJECT (self, "Configuring to use upstream buffers ...");
+
+  for (i = 0; i < n; i++) {
+    GstOMXBuffer *buf = g_ptr_array_index (mem_port->buffers, i);
+    buffers = g_list_append (buffers, buf->omx_buf->pBuffer);
+    GST_LOG_OBJECT (self, "Adding buffer %p to the use list",
+        buf->omx_buf->pBuffer);
+  }
+
+  ret = gst_omx_video_enc_component_init (self, buffers);
+
+  g_list_free (buffers);
+
+  return ret;
+
+reconfigure_error:
```

Comment:
This _gotos_ may not be tabbed/spaced properly.

Reply:
I think it's fine, I have always seen them without any tabs or spaces at the beginning of the gotos tags.